src/pages/home/sidebar/SidebarLinksData.tsx

```diff
@@ -1,149 +1,133 @@
-
-                /** The transaction violation type */
-                type: PropTypes.string,
-
-                /** The transaction violation message */
-                message: PropTypes.string,
-
-                /** The transaction violation data */
-                data: PropTypes.shape({
-                    /** The transaction violation data field */
-                    field: PropTypes.string,
-
-                    /** The transaction violation data value */
-                    value: PropTypes.string,
-                }),
-            }),
-        ),
-    }),
+    transactionViolations: OnyxCollection<OnyxTypes.TransactionViolations>;
 
-    ...withCurrentUserPersonalDetailsPropTypes,
+    /** All policy members */
+    policyMembers: OnyxCollection<OnyxTypes.PolicyMembers>;
 };
 
-const defaultProps = {
-    chatReports: {},
-    isLoadingApp: true,
-    priorityMode: CONST.PRIORITY_MODE.DEFAULT,
-    betas: [],
-    policies: {},
-    policyMembers: {},
-    transactionViolations: {},
-    allReportActions: {},
-    ...withCurrentUserPersonalDetailsDefaultProps,
-};
+type SidebarLinksDataProps = CurrentReportIDContextValue &
+    SidebarLinksDataOnyxProps & {
+        /** Toggles the navigation menu open and closed */
+        onLinkClick: () => void;
+
+        /** Safe area insets required for mobile devices margins */
+        insets: EdgeInsets | undefined;
+    };
 
 function SidebarLinksData({
-    isFocused,
     allReportActions,
     betas,
     chatReports,
-    currentReportID,
     insets,
-    isLoadingApp,
+    isLoadingApp = true,
     onLinkClick,
     policies,
-    priorityMode,
-    network,
+    priorityMode = CONST.PRIORITY_MODE.DEFAULT,
     policyMembers,
     transactionViolations,
-    currentUserPersonalDetails,
-}) {
+    currentReportID,
+}: SidebarLinksDataProps) {
+    const {accountID} = useCurrentUserPersonalDetails();
+    const network = useNetwork();
+    const isFocused = useIsFocused();
     const styles = useThemeStyles();
     const {activeWorkspaceID} = useActiveWorkspace();
     const {translate} = useLocalize();
     const prevPriorityMode = usePrevious(priorityMode);
-
-    const policyMemberAccountIDs = getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, currentUserPersonalDetails.accountID);
+    const policyMemberAccountIDs = getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, accountID);
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    useEffect(() => Policy.openWorkspace(activeWorkspaceID, policyMemberAccountIDs), [activeWorkspaceID]);
+    useEffect(() => Policy.openWorkspace(activeWorkspaceID ?? '', policyMemberAccountIDs), [activeWorkspaceID]);
 
-    const reportIDsRef = useRef(null);
+    const reportIDsRef = useRef<string[] | null>(null);
     const isLoading = isLoadingApp;
-    const optionListItems = useMemo(() => {
+    const optionListItems: string[] | null = useMemo(() => {
```

Comment:
```suggestion
    const optionListItems: string[] = useMemo(() => {
```
to make it work, you need to update this check to narrow down the type -> it should be ok since reportIDsRef.current can be equal to reportIDs only if it's not null
```
if (reportIDsRef.current && deepEqual(reportIDsRef.current, reportIDs)) {
            return reportIDsRef.current;
}
```
or return reportIDs -> it should be ok since we check that it's equal to reportIDsRef.current
```
if (deepEqual(reportIDsRef.current, reportIDs)) {
            return reportIDs;
}
```